src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,57 @@
+package duke;
+
+/**
+ * Represents Parser class that parses input commands.
+ */
+public class Parser {
+
+    /**
+     * Returns a specific command after parsing.
+     *
+     * @param userInput User input as a String.
+     * @return Specific input command.
+     * @throws DukeException When invalid command or formatting is entered.
+     */
+    public static Command parseCommands(String userInput) throws DukeException {
+        CommandEnum command;
+        try {
+            String[] input = userInput.split(" ");
+            int len = userInput.length();
+            String extractCommand = input[0];
+            command = CommandEnum.valueOf(extractCommand.toUpperCase());
+        } catch (NullPointerException | IllegalArgumentException ex) {
+            System.out.println(Ui.getLine());
+            System.out.println(Ui.getBot());
+            throw new DukeException(
+                    "You have keyed in an invalid command or formatting!\n" +
+                            "(Valid commands: todo, deadline, event, list," +
+                            " delete, bye, done, find)");
+        }
+
+        switch (command) {
+            case BYE:
```

Comment:
Perhaps can consider checking the coding convention for switch cases, I recall that case should be aligned with the switch command in terms of spacing. 👍

Reply:
Just noticed it! Thank you for pointing it out :-)